cloudevents-sdk-rdkafka/src/kafka_consumer_record.rs

```diff
@@ -0,0 +1,281 @@
+use crate::headers;
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Encoding, MessageAttributeValue, MessageDeserializer,
+    Result, StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::{message, Event};
+use rdkafka::message::{BorrowedMessage, Headers, Message, OwnedMessage};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::str;
+
+pub struct ConsumerRecordDeserializer {
+    pub(crate) headers: HashMap<String, Vec<u8>>,
+    pub(crate) payload: Vec<u8>,
+}
+
+impl ConsumerRecordDeserializer {
+    pub fn owned_new(message: OwnedMessage) -> ConsumerRecordDeserializer {
+        let mut resp_des = ConsumerRecordDeserializer {
+            headers: HashMap::new(),
+            payload: Vec::new(),
+        };
+        let headers = message.headers().unwrap();
```

Comment:
is this `unwrap()` safe? should we return an error in case no headers are there?

Reply:
yeah it seems headers are more of a feature in kafka than a necessity.